.ci/compute_projects.py

```diff
@@ -0,0 +1,186 @@
+# Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+"""Computes the list of projects that need to be tested from a diff.
+
+Does some things, spits out a list of projects.
+"""
+
+from collections.abc import Set
+import pathlib
+import platform
+import sys
+
+PROJECT_DEPENDENCIES = {
```

Comment:
Can you add a comment about the content of this? What is this conveying and when you a project include others in its list?

Reading the usage of this, it seems like it is handling transitive dependencies for example, so I assume we should only list direct dependencies?

Also some dependencies are optional (depending on the CMake configuration), so worth giving some idea of what is the intent and the effect (to know when / whether to be conservative or how to exercise judgement).

Reply:
I added a comment.

Yes, it handles transitive dependencies automatically (and I made sure to note that in the comment).

The dependencies should match what is required for the configuration int the premerge CI (also mentioned that in the comment).